modules/build/src/main/scala/scala/build/input/Inputs.scala

```diff
@@ -263,14 +263,25 @@ object Inputs {
     stdinOpt: => Option[Array[Byte]],
     acceptFds: Boolean,
     enableMarkdown: Boolean,
-    isRunWithShebang: Boolean
+    programInvokeData: ScalaCliInvokeData
   ): Seq[Either[String, Seq[Element]]] = args.zipWithIndex.map {
     case (arg, idx) =>
-      lazy val path      = os.Path(arg, cwd)
-      lazy val dir       = path / os.up
-      lazy val subPath   = path.subRelativeTo(dir)
-      lazy val stdinOpt0 = stdinOpt
-      lazy val content   = os.read.bytes(path)
+      lazy val path             = os.Path(arg, cwd)
+      lazy val dir              = path / os.up
+      lazy val subPath          = path.subRelativeTo(dir)
+      lazy val stdinOpt0        = stdinOpt
+      lazy val content          = os.read.bytes(path)
+      val isRunWithShebang      = programInvokeData.subCommand == SubCommand.Shebang
+      val isRunWithDefault      = programInvokeData.subCommand == SubCommand.Default
```

Comment:
having 2 separate vals for this seems verbose, if it's run with shebang, it's definitely not run with default and vice versa...
if you really want the descriptive names to be used here, you can make `isRunWithShebang` and `isRunWithDefault` methods of `ScalaCliInvokeData`.
or just do the comparison on the fly, it's self-descriptive